utils/GRI_bed_filter/version1_gene-only/gri_v1.R

```diff
@@ -0,0 +1,248 @@
+    paste0("bed file coverage is: ", size.per * 100, "% ;"),
+    bed.size,
+    "bp out of",
+    seq.size,
+    "bp"
+  ))
+}
+
+# 1. include gencode v46 annotation ----------------------------
+
+## convert gtf data to data.table and add chr size
+gtf_dt <- as.data.table(gtf_data)
+gtf_dt_size <- merge(gtf_dt, chr_size, by = "seqnames") # no MT chr
+
+## select protein coding genes
+genes <- gtf_dt_size[type == "gene" & gene_type == "protein_coding"]
+
+## extract gene location info and save into bed format
+## add source column
+bed_coding <- genes |>
+  dplyr::select(
+    chr = seqnames,
+    start = start,
+    end = end,
+    name = gene_name,
+    strand = strand,
+  ) |>
+  mutate(source = "GencodeV46.protein_coding")
+
+## merge overlapped region
+gr <- GRanges(
+  seqnames = bed_coding$chr,
+  ranges = IRanges(start = bed_coding$start, end = bed_coding$end)
+)
+
+merged_gr <- reduce(gr)
+bed_coding.dt <- as.data.table(merged_gr)
+coverage_bed(bed_coding.dt)
+
+# 2. include hgmd 2022 variants -----------------------------------------
+
+## extract fields and convert to data.table
+info_dt <- as.data.table(info(hgmd_data))
+range_dt <- as.data.table(rowRanges(hgmd_data))
+
+## merged useful columns
+hgmd <- cbind(range_dt, info_dt) |>
+  dplyr::select(seqnames, start, end, GENE, STRAND, CLASS, MUT) |>
+  mutate(source = "HGMD_2022")
+
+## exclude "rejected variants" in HGMD database
+hgmd_noR <- hgmd[-which(hgmd$CLASS == "R"), ]
+hgmd_noR <- hgmd_noR |>
+  dplyr::mutate(seqnames = paste0("chr", seqnames))
+
+## add 50bp flanking region
+hgmd_noR <- merge(hgmd_noR, chr_size, by = "seqnames")
+hgmd_noR_flank <- hgmd_noR |>
+  dplyr::mutate(
+    final_start = ifelse(start > 50, start - 50, 0),
+    final_end = ifelse((end + 50) > chr_end, chr_end, end + 50),
+    .after = seqnames
+  )
+
+## generate bed
+bed_hgmd <- hgmd_noR_flank |>
+  dplyr::select(
+    chr = seqnames,
+    start = final_start,
+    end = final_end,
+    name = GENE,
+    strand = STRAND,
+    source
+  )
+
+## merge bed files from step1 and step2, and reduce them into non-overlapping regions
+bed_gencode_hgmd <- rbind(bed_coding, bed_hgmd, fill = TRUE) |>
+  arrange(chr, start, end)
+
+gr <- GRanges(
+  seqnames = bed_gencode_hgmd$chr,
+  ranges = IRanges(start = bed_gencode_hgmd$start, end = bed_gencode_hgmd$end)
+)
+
+merged_gr <- reduce(gr)
+bed_gencode_hgmd.dt <- as.data.table(merged_gr)
+coverage_bed(bed_gencode_hgmd.dt)
+
+# 3. Include clinvar variants----------------------------------------
+
+## obtain pathogenic variants
+info_df <- as.data.frame(clinvar_data@info) #faster to load as dataframe than datatable
+range_df <- as.data.frame(clinvar_data@rowRanges)
+clinvar <- cbind(range_df, info_df)
+clinvar_patho <- clinvar |>
+  filter(grepl("pathogenic", CLNSIG))
+
+## remove MT chromosome
+## rename seqnames
+clinvar_patho <- clinvar_patho |>
+  filter(seqnames != "MT")
+clinvar_patho <- clinvar_patho |>
+  mutate(seqnames = paste0("chr", seqnames))
```

Comment:
We don't use `chr` prefix for the chromosomes, will it be in the final output? it will be fine if it can be used - can you test if the filter is working without any problem? If so, we can keep it, otherwise we need to make some change.

Reply:
@hyunhwan-bcm, In current implementation, we remove chr prefixes, so no need to worry about.

Reply:
okay, thanks for checking no need to address this.